includes/functions.php

```diff
@@ -2294,4 +2319,111 @@ function tsml_date_localised($format, $timestamp = null)
 	$datetime->setTimezone(new DateTimeZone($timezone));
 	return $datetime->format($format);
 }
+
+
+//function:	return array of updated feed records only when difference detected between matched feed record and the local db record
+function tsml_get_import_changes_only($feed_meetings, $data_source_url, $data_source_last_update, &$db_ids_to_delete, &$message_lines)
+{
+	$db_meetings = $feed_slugs = $meetings_updated = [];
+	$week_days	= [
+		__('Sunday', '12-step-meeting-list'),
+		__('Monday', '12-step-meeting-list'),
+		__('Tuesday', '12-step-meeting-list'),
+		__('Wednesday', '12-step-meeting-list'),
+		__('Thursday', '12-step-meeting-list'),
+		__('Friday', '12-step-meeting-list'),
+		__('Saturday', '12-step-meeting-list'),
+	];
+
+	// get local meetings 
+	$all_db_meetings = tsml_get_meetings();
+	$ds_ids = tsml_get_data_source_ids($data_source_url);
+	sort($ds_ids);
+
+	/* filter out all but the data source meetings  */
+	foreach ($all_db_meetings as $db_meeting) {
+		$db_id = $db_meeting['id'];
+		if (in_array($db_id, $ds_ids)) {
+			array_push($db_meetings, $db_meeting);
+		}
+	}
```

Comment:
i think using `array_filter` might be a nice improvement here, instead of making a new array with `foreach`

Reply:
Not sure how that would work. This might be a teachable moment. Could I get an example of how that would look?

Reply:
Sure so instead of having both `$db_meetings` and `$all_db_meetings` and the repeat loop, you could do it like this:

```
$db_meetings = array_filter(tsml_get_meetings(), function ($meeting) {
  return in_array($meeting['id'], $ds_ids);
});
```

`array_filter` loops through the array (in this case`tsml_get_meetings`) and uses a callback function to decide whether to filter each item in or out

Reply:
The array_filter() function filters the values of a single array using the callback function. I'm processing two arrays: get_meetings and ds_ids. So the above code passes the meetings array, but the ds_ids array is undefined.  I did get it to work when I globalized the ds_ids array (which I renamed to $tsml_data_source_ids). The code that worked for me: 	**$db_meetings = array_filter(tsml_get_meetings(), function ($meeting) { global $tsml_data_source_ids;  return in_array($meeting['id'], $tsml_data_source_ids); });**